modules/database/funcs.go

```diff
@@ -135,6 +137,9 @@ func (d *Database) CreateOrPatchDB(
 			return err
 		}
 
+		// If the service object doesn't have our finalizer, add it.
+		controllerutil.AddFinalizer(instance, h.GetFinalizer())
```

Comment:
we want to set the finalizer on the `db` CR, not on the owner object. so I think it should be `controllerutil.AddFinalizer(db, h.GetFinalizer())` instead

Reply:
ok

Reply:
ok